pyboy/core/ram.py

```diff
@@ -15,16 +18,21 @@
 
 
 class RAM:
-    def __init__(self, random=False):
-        if random: # NOTE: In real life, the RAM is scrambled with random data on boot.
-            raise Exception("Random RAM not implemented")
-
-        self.internal_ram0 = array.array("B", [0] * (INTERNAL_RAM0))
-        self.non_io_internal_ram0 = array.array("B", [0] * (NON_IO_INTERNAL_RAM0))
-        self.io_ports = array.array("B", [0] * (IO_PORTS))
-        self.internal_ram1 = array.array("B", [0] * (INTERNAL_RAM1))
-        self.non_io_internal_ram1 = array.array("B", [0] * (NON_IO_INTERNAL_RAM1))
-        self.interrupt_register = array.array("B", [0] * (INTERRUPT_ENABLE_REGISTER))
+    def __init__(self, randomize=False):
+        typecode = "B"
+
+        self.internal_ram0 = array.array(typecode, [0] * (INTERNAL_RAM0))
+        self.non_io_internal_ram0 = array.array(typecode, [0] * (NON_IO_INTERNAL_RAM0))
+        self.io_ports = array.array(typecode, [0] * (IO_PORTS))
+        self.internal_ram1 = array.array(typecode, [0] * (INTERNAL_RAM1))
+        self.non_io_internal_ram1 = array.array(typecode, [0] * (NON_IO_INTERNAL_RAM1))
+        self.interrupt_register = array.array(typecode, [0] * (INTERRUPT_ENABLE_REGISTER))
+
+        if randomize: # NOTE: In real life, the RAM is scrambled with random data on boot.
+            for a in (self.internal_ram0, self.non_io_internal_ram0, self.io_ports, self.internal_ram1,
+                      self.non_io_internal_ram1, self.interrupt_register):
+                for i in range(len(a)):
+                    a[i] = random.randrange(REGISTER_MAX + 1)
```

Comment:
I would suggest using random.getrandbits(8) instead. Also, IO should not be randomized.

Reply:
I like this. It is accurate, concise and the intentions are clear